### PR TITLE
triqs: 1.4 (new formula)

### PR DESCRIPTION
**Pull request by Wentzell**
TRIQS (Toolbox for Research on Interacting Quantum Systems) is a scientific project providing a set of C++ and Python libraries to develop new tools for the study of interacting quantum systems.

The goal of this toolkit is to provide high level, efficient and simple to use libraries in C++ and Python, and to promote the use of modern programming techniques.

TRIQS is free software distributed under the GPL license.

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [x] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [ ] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

### CODE REVIEW

**Comment by Wentzell** [comment]
Seems to fail for the same reasons discussed already in #5065

**Comment by jonchang** [comment]
@BrewTestBot test this please

File: triqs.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class Triqs < Formula
+  desc "Applications and Libraries for Physics Simulations"
+  homepage "https://triqs.ipht.cnrs.fr/"
+
+  head "https://github.com/TRIQS/triqs.git"
+
+  stable do
+    url "https://github.com/TRIQS/triqs/archive/1.4.tar.gz"
+    sha256 "98378d5fb934c02f710d96eb5a3ffa28cbee20bab73b574487f5db18c5457cc4"
+  end
```
Comment (jonchang) [inline triqs.rb]: This can just be `url "https://...."`

File: triqs.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  option "with-test", "Build and run shipped tests"
```
Comment (jonchang) [inline triqs.rb]: Should these tests be built and run by default?

File: triqs.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  depends_on "cmake" => :build
+  depends_on :mpi => :cxx
+  depends_on "boost"
+  depends_on "hdf5"
+  depends_on "fftw"
+  depends_on "gmp"
+  depends_on "python"
```
Comment (jonchang) [inline triqs.rb]: `depends_on :python` unless there's a good reason to require Homebrew's version of Python

File: triqs.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  depends_on "pkg-config" => :run
+
+  def install
+    system "pip", "install", "--upgrade", "pip"
+    system "pip", "install", "--upgrade", "mako"
+    system "pip", "install", "--upgrade", "scipy"
+    system "pip", "install", "--upgrade", "numpy"
+    system "pip", "install", "--upgrade", "jupyter"
+    system "pip", "install", "--upgrade", "--force-reinstall", "--no-binary=h5py", "h5py"
+    system "pip", "install", "--upgrade", "--force-reinstall", "--no-binary=mpi4py", "mpi4py"
```
Comment (jonchang) [inline triqs.rb]: See `circlator.rb` or `anvio.rb` for examples on how to install python package dependencies

File: triqs.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      ..
+      -DCMAKE_BUILD_TYPE=Release
+      -DCMAKE_INSTALL_PREFIX=#{prefix}
+    ]
+    args << ("-DBuild_Tests=" + (build.with?("test") ? "ON " : "OFF "))
+
+    mkdir "build" do
+      rm("../test/triqs/statistics/autocorrelation_jackknife.cpp") if build.stable?
+      system "cmake", *args
+      system "make", "-j", "1"
```
Comment (jonchang) [inline triqs.rb]: `ENV.deparallelize do ... end`

File: triqs.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      system "make", "test" if build.with? "test"
+      system "make", "install"
+    end
+  end
+
+  def post_install
+    chmod 0555, bin/"clang_parser.py"
```
Comment (jonchang) [inline triqs.rb]: Shouldn't this be in the `install` step?

File: triqs.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
+class Triqs < Formula
+  desc "Applications and Libraries for Physics Simulations"
```
Comment (jonchang) [inline triqs.rb]: Wouldn't "Toolbox for Research on Interacting Quantum Systems" be more appropriate?

File: triqs.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  homepage "https://triqs.ipht.cnrs.fr/"
+
+  head "https://github.com/TRIQS/triqs.git"
+
+  stable do
+    url "https://github.com/TRIQS/triqs/archive/1.4.tar.gz"
+    sha256 "98378d5fb934c02f710d96eb5a3ffa28cbee20bab73b574487f5db18c5457cc4"
+  end
+
```
Comment (jonchang) [inline triqs.rb]: Please include a `# tag` and `# doi` comment to reference the manuscript. See `raxml.rb` for an example.

File: triqs.rb (hunk 7 above)
Comment (Wentzell) [reply triqs.rb]: Agreed! Added that two my local branch.

File: triqs.rb (hunk 1 above)
Comment (Wentzell) [reply triqs.rb]: Ok, made these changes in my local branch for now.

File: triqs.rb (hunk 8 above)
Comment (Wentzell) [reply triqs.rb]: Added.

File: triqs.rb (hunk 2 above)
Comment (Wentzell) [reply triqs.rb]: Yes they should be run when the package is built from source. We are now adding them to the install procedure.

File: triqs.rb (hunk 5 above)
Comment (Wentzell) [reply triqs.rb]: Added.

File: triqs.rb (hunk 6 above)
Comment (Wentzell) [reply triqs.rb]: Agreed and added!

File: triqs.rb (hunk 3 above)
Comment (Wentzell) [reply triqs.rb]: Thanks for the comment! Still looking into this...

File: triqs.rb (hunk 4 above)
Comment (Wentzell) [reply triqs.rb]: Thanks for the comment! Still looking into this...

File: triqs.rb (hunk 3 above)
Comment (Wentzell) [reply triqs.rb]: I adjusted to formula to work also with the MacOs system python

File: triqs.rb (hunk 4 above)
Comment (Wentzell) [reply triqs.rb]: I changed to formula to install most of the python dependencies via virtualenv into libexec/venv

**Comment by Wentzell** [comment]
Updated the formula according to the requested changes.
@jonchang Please review

**Comment by Wentzell** [comment]
It seems that the permission changes in the end of the formula are not properly working. 
```
chmod 0555, bin/"clang_parser.py"
chmod 0555, bin/"cpp2doc_tools.py"
```
Why could that be?

**Comment by iMichka** [comment]
We updated our "new formula" policy for homebrew-science:

> We will no longer accept new formula in homebrew-science, as this tap is being phased out. Please consider submitting your formulae to https://github.com/Homebrew/homebrew-core or host it in your own tap (https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.html).

I am closing this pull request, if you need any advice please feel free to add a comment in this issue. Thanks!

See also #6365 for the discussion about what is going to happen with homebrew-science.

**Pull request closed without merging**